src/NearConfig.ts

```diff
@@ -18,12 +25,65 @@ export const nearConfigMainnet = {
   headers: {},
 };
 
+export function getConnectionConfig(chain: Chain, network: Network, shardName = "", apiKey = ""): any {
+  if (chain === Chain.Near) {
+    return getNearConfig(network);
+  }
+  if (chain === Chain.Calimero) {
+    const networkId = getNetworkId(chain, network, shardName)
+    return getCalimeroConfig(networkId, shardName, apiKey);
+  }
+  return null;
+}
+
 export function getNearConfig(network: Network): any {
   if (network === Network.Testnet) return nearConfigTestnet;
   if (network === Network.Mainnet) return nearConfigMainnet;
   return null;
 }
 
-export function getRpcUrl(network: Network): string {
+export function getCalimeroConfig(networkId: string, shardName: string, apiKey: string): any {
+  return {
+    networkId: networkId,
+    nodeUrl: getCalimeroRpcUrl(shardName),
+    headers: {
+      "x-api-key": apiKey,
+    }
+  };
+}
+
+export function getNearRpcUrl(network: Network): string {
   return getNearConfig(network).nodeUrl;
 }
+
+// TODO: extract parts of this URL to constants
+export function getCalimeroRpcUrl(shardName: string): string {
```

Comment:
Calimero specific should probably not reside in NearConfig, either name this NetworkConfig or move into its own file

Reply:
I renamed this file to NetworkConfig as proposed